examples/swe_agent/main.ipynb

```diff
@@ -0,0 +1,304 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Example with SWE-agent\n",
+    "\n",
+    "SWE-agent(SoftWare Engineering Agent) is an agent designed for solving real world software engineering problems, such as fixing github issues.\n",
+    "More details can be found in their [homepage](https://swe-agent.com/) and related [github repo](https://swe-agent.com/).\n",
+    "\n",
+    "In the example here, we partially implement the SWE-agent, and provide a simple example of how to use the implemented SWE-agent to fix a bug in a python file.\n",
+    "You should note that currently how to enable agents with stronger programming capabilities remains an open challenge, and the performance of the paritially implemented SWE-agent is not guaranteed.\n",
+    "\n",
+    "## Prerequisites\n",
+    "\n",
+    "- Follow [READMD.md](https://github.com/modelscope/agentscope) to install AgentScope. \n",
+    "- Prepare a model configuration. AgentScope supports both local deployed model services (CPU or GPU) and third-party services. More details and example model configurations please refer to our [tutorial]\n",
+    "- Understand the ServiceFactory module and how to use it to pre-process the tool functions for LLMs. You can refer to the [ReAct agent example](../conversation_with_react_agent/main.ipynb).\n",
```

Comment:
1. Service toolkit module rather than service factory module.
2. Add link to tutorial (service section, https://modelscope.github.io/agentscope/en/tutorial/204-service.html)